qiskit/converters/circuit_to_gate.py

```diff
@@ -80,10 +81,15 @@ def find_bit_position(bit):
 
     if gate.num_qubits > 0:
         q = QuantumRegister(gate.num_qubits, 'q')
+    if gate.num_clbits > 0:
+        c = ClassicalRegister(gate.num_clbits, 'c')
 
-    definition = list(map(lambda x:
-                          (x[0], list(map(lambda y: q[find_bit_position(y)], x[1]))),
-                          definition))
-    gate.definition = definition
+    out_definition = []
+    for x in definition:
+        x_def = (x[0], list(map(lambda y: q[find_bit_position(y)], x[1])),
+                 list(map(lambda z: c[find_bit_position(z)], x[2])))
```

Comment:
It might be clearer to hard code this to `[]` with a comment that op cregs are precluded by the `if any(not isinstance(op, Gate) for op in circ.data)` check above.

Reply:
That's fair, I was blindly just trying to mirror the behavior from qregs for cregs when I wrote this and didn't think through the implications until I was writing the commit message :) I'll respin this now

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3555/commits/5f55a28f366afee935170d707ae913e2d785b8af